cmake/Modules/Packages.cmake

```diff
@@ -450,17 +450,26 @@ endif()
 
 #set(DEV_WARNINGS ${CMAKE_SUPPRESS_DEVELOPER_WARNINGS})
 
-checkout_git_submodule(RECURSIVE
-    RELATIVE_PATH external/cereal
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    REPO_URL https://github.com/jrmadsen/cereal.git
-    REPO_BRANCH timemory)
+if (TIMEMORY_BUILD_CEREAL)
```

Comment:
Again, can't use external cereal

Reply:
> I avoid CMake 2 semantics pretty strongly, e.g. `add_definitions(...)`, `add_compile_options` in favor of the target-based semantics. All global compiler flags should be added to `timemory-compile-options`, e.g. `target_compile_definitions(timemory-compile-options INTERFACE <definitions>)`.

Yes, I agree. (I did not really understand that timemory-comile-options is the target to attach compiler flags to.) I will update this.

> Are you using cereal? The library isn't supported upstream and I have a custom copy bc they won't accept (or even respond to my PRs to get the `MinimalJSONOutputArchive` into the repo which contains no line-breaks, etc.) so an external cereal will break it until I move it into a private copy

Yes, I use cereal - I depend on cereal, but I have not needed any changes to it. Where is your pull request that they have ignored? Is there a way I can ping them to at least respond? (I am concerned that your impression is that cereal is no longer being maintained.) As for the "external build of cereal," what I did is simply use your version of cereal but I need it to be installed globally independent of timemory. So I am simply using your fork rather than the "official" fork, but I do not want the timemory build to install it. I have already installed it (because I am using it outside of the timemory usage). Does this make sense?

Reply:
Sure it makes sense. The [PR was here](https://github.com/USCiLab/cereal/pull/613). I pinged them, other people pinged them. No response for a month or two. Once I started making a few more edits, like creating a dummy `makeArray()` for XML, I just decided to close it bc I didn't want to maintain multiple branches in my fork. In your build you can link to the `timemory::timemory-cereal` Interface library as if it were globally installed but if you are using it for other projects, I understand the inclination. Unfortunately though I'd prefer to just maintain an internal version at this point rather than have the CMake code _imply_ that any standard cereal installation will suffice, if that makes sense. If I were to move it into an internal build it would be basically like what cereal has with rapidjson -- it's there but unless you are looking for it, you don't really realize it's there.

Reply:
Ah yeah, I waited almost 6 months for a reply on that PR (opened in January, closed in June)